formver.compiler-plugin/core/src/org/jetbrains/kotlin/formver/core/conversion/StmtConversionContext.kt

```diff
@@ -269,7 +272,10 @@ fun StmtConversionContext.convertFunctionWithBody(
     val body = extractReturnedExprFromPureFunctionBody(bodyWithPosition, declaration.source)
     val pureLinearizer = PureLinearizer(declaration.source)
 
-    // TODO: Hook in purity check here
+    if (!body.isPure()) throw SnaktInternalException(
```

Comment:
Worth noting: this is also not an internal exception, it's a user error.  But it's fine to leave that as a TODO.